tests/integration/test_logcollector/test_macos/test_macos_file_status_predicate.py

```diff
@@ -0,0 +1,107 @@
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_macos_file_status_predicate.yaml')
+
+parameters = [{'ONLY_FUTURE_EVENTS': 'yes'}, {'ONLY_FUTURE_EVENTS': 'no'}]
+metadata = [{'only-future-events': 'yes'}, {'only-future-events': 'no'}]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f'{x["ONLY_FUTURE_EVENTS"]}' for x in parameters]
+
+# Maximum waiting time in seconds to find the logs on ossec.log
+wazuh_log_monitor_timeout = 30
+
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = {'logcollector.vcheck_files': str(file_status_update_time)}
+
+
+@pytest.fixture(scope='module')
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+def extra_configuration_before_yield():
+    """Delete file status file."""
+    os.remove(LOGCOLLECTOR_FILE_STATUS_PATH) if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH) else None
+
+
+def callback_log_bad_predicate(line):
+    match = re.match(r'.*Execution error \'log:', line)
+    if match:
+        return True
+    return None
+
+
+def callback_log_exit_log(line):
+    match = re.match(r'.*macOS \'log stream\' process exited, pid:', line)
+    if match:
+        return True
+    return None
+
+
+# Fixtures
+@pytest.fixture(scope='module', params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def get_connection_configuration():
+    """Get configurations from the module."""
+    return DEFAULT_AUTHD_REMOTED_SIMULATOR_CONFIGURATION
+
+
+def test_macos_file_status_predicate(get_local_internal_options, configure_local_internal_options, get_configuration,
+                                     configure_environment, get_connection_configuration, init_authd_remote_simulator,
+                                     restart_logcollector):
+
+    """Checks that logcollector does not store 'macos'-formatted localfile data since its predicate is erroneous.
+
+    The agent is connected to the authd simulator and uses a dummy localfile (/Library/Ossec/logs/active-responses.log)
+    which triggers the creation of file_status.json
+
+    Raises:
+        TimeoutError: If the callbacks, that checks the expected logs, are not satisfied in the expected time.
+        FileNotFoundError: If the file_status.json is not available in the expected time.
+    """
+
+    wazuh_log_monitor.start(timeout=wazuh_log_monitor_timeout,
+                            callback=callback_log_bad_predicate,
+                            error_message='Expected log that matches the regex '
+                                          '".*Execution error \'log:" could not be found')
```

Comment:
```suggestion
                            error_message="Expected log that matches the regex "
                                          "'.*Execution error \'log:' could not be found")
```

Reply:
It wouldn't work, because the message expects the character to be a double quotation mark